server/models/User.js

```diff
@@ -351,7 +351,8 @@ export default (Sequelize, DataTypes) => {
 
         let update = false;
         const allowedFields = 
-          [ 'firstName',
+          [ 'username',
```

Comment:
If there's another username, it'll pop error message back to website. Problem is if there is a group slug, it won't error out but the profile will become inaccessible because we prioritize group slugs over usernames if there is a clash.

I check for it in userlib/suggestUsername but that only gets called if User doesn't have a username. hmm...

The right solution is to create a new table that stores slugs and usernames and has a unique index. 

One quick and dirty solution would be to pull out the code from userlib that checks both tables and call that in beforeCreate hook of User model?